NEXT-12794 - [NEW] Administration: Extend webpack configuration (#199)
shopware/docs: guides/plugins/plugins/administration/extending-webpack.md

# Extending Webpack

## Overview

The Shopware 6 Administration uses [Webpack](https://webpack.js.org/) as a static module bundler.
The default configuration should work for most plugins, 
but if you really need to extend the webpack configuration,
this guide will show you how it's done.

## Extending the Webpack configuration

> The Webpack configuration can be extended by creating the file `<plugin root>src/Resources/app/administration/build/webpack.config.js`

Reviewer: ```suggestion
The Webpack configuration can be extended by creating the file `<plugin root>/src/Resources/app/administration/build/webpack.config.js`
```

> and exporting a function from it, that returns an [webpack configuration object](https://webpack.js.org/configuration/). 

Reviewer 2: ```suggestion
and exporting a function from it. This will return a [webpack configuration object](https://webpack.js.org/configuration/),
```

> As seen below:

Reviewer 2: ```suggestion
as seen below:
```

> {% code title="<plugin root>src/Resources/app/administration/build/webpack.config.js" %}

Reviewer: ```suggestion
{% code title="<plugin root>/src/Resources/app/administration/build/webpack.config.js" %}
```

```javascript
const path = require('path');

module.exports = () => {
    return {
        resolve: {
            alias: {
                SwagBasicExample: path.join(__dirname, '..', 'src')
            }
        }
    };
};
```
{% endcode %}

> The configuration is then automatically loaded and then merged with the Shopware provided webpack configuration and all other plugin webpack configurations.

Reviewer 2: ```suggestion
This way, the configuration is automatically loaded and then merged with the Shopware provided webpack configuration, including all other plugin webpack configurations.
```

Merging is done with the [webpackMerge](https://github.com/survivejs/webpack-merge) library.

> This merging makes it technically possible to override the Shopware provided configuration, although it is generally advised against.

Reviewer: use `hint` block 
```suggestion

{% hint style="danger" %}
This merging makes it technically possible to override the Shopware provided configuration, although it is generally advised against.
{% endhint %}
```

# Extending Webpack

## Overview

The Shopware 6 Administration uses [Webpack](https://webpack.js.org/) as a static module bundler.

> The default configuration should work for most plugins, 

Reviewer 3: I would say that if you don´t have needs for any special changes then you don´t need to bother with webpack at all. Your plugin will just work.

## Extending the Webpack configuration

The Webpack configuration can be extended by creating the file `<plugin root>/src/Resources/app/administration/build/webpack.config.js`
and exporting a function from it. This will return a [webpack configuration object](https://webpack.js.org/configuration/),
as seen below:

{% code title="<plugin root>/src/Resources/app/administration/build/webpack.config.js" %}
```javascript
const path = require('path');

module.exports = () => {
    return {
        resolve: {
            alias: {
                SwagBasicExample: path.join(__dirname, '..', 'src')
            }
        }
    };
};
```
{% endcode %}

This way, the configuration is automatically loaded and then merged with the Shopware provided webpack configuration, including all other plugin webpack configurations.
Merging is done with the [webpackMerge](https://github.com/survivejs/webpack-merge) library.

{% hint style="danger" %}
This merging makes it technically possible to override the Shopware provided configuration, 

> but you shouldn't change the default configurations output,

Reviewer 3: ```suggestion
but you shouldn't change the default configurations so that the builded core code will differ from the original builded code without your configuration,
```

> This merging makes it technically possible to override the Shopware provided configuration, although it is generally advised against.

Reviewer 3: It is not `it is generally advised against`. Its possible if your changes don´t change the core code output. If it will change the bundled code of the core then your plugin could potentially not work
